### PR TITLE
io_uring: add prep functions for current working directory

### PR DESCRIPTION
**Pull request by uarif1**
This is done for unlink, rename, mkdir, symlink and link.

Signed-off-by: Usama Arif <usama.arif@bytedance.com>

### CODE REVIEW

**Comment by axboe** [comment]
Thanks for doing this. Two minor comments:

- Shouldn't the test/rename.c change itself actually use io_uring_prep_rename()?
- Can you do the man page additions as well? They should just go into the existing *at() man pages, no need for a separate page for them

File: src/include/liburing.h [hunk 1]
```diff
@@ -739,6 +747,15 @@ static inline void io_uring_prep_renameat(struct io_uring_sqe *sqe, int olddfd,
 	sqe->rename_flags = (__u32) flags;
 }
 
+static inline void io_uring_prep_rename(struct io_uring_sqe *sqe,
```
Comment (axboe) [inline src/include/liburing.h:750]: I also think it'd be cleaner if this one just called io_uring_prep_renameat(), and ditto for the other ones. Then we keep the RENAMEAT sqe fields consistent by using the same parent helper.

**Comment by YoSTEALTH** [comment]
wouldn't it be better to call `io_uring_prep_rw` directly? this way it actually makes it worth using `io_uring_prep_mkdir`


```c
static inline void io_uring_prep_mkdir(struct io_uring_sqe *sqe,
					const char *path, mode_t mode)
{
    io_uring_prep_rw(IORING_OP_MKDIRAT, sqe, AT_FDCWD, path, mode, 0);
}
```

vs

```c
static inline void io_uring_prep_mkdir(struct io_uring_sqe *sqe,
					const char *path, mode_t mode)
{
	io_uring_prep_mkdirat(sqe, AT_FDCWD, path, mode);
}
```

**Comment by ammarfaizi2** [comment]
> wouldn't it be better to call `io_uring_prep_rw` directly? this way it actually makes it worth using `io_uring_prep_mkdir`

How come it be better? Those are the same thing. They are all inline anyway.

**Comment by YoSTEALTH** [comment]
> How come it be better? Those are the same thing. They are all inline anyway.

Right, I overlooked it being `inline`.

**Pull request merged**